### ISSUE TITLE
PWA not available

### ISSUE THREAD

**Issue by benmccann** [photoprism/photoprism#852]
I've seen numerous references that PhotoPrism is a PWA. However, I cannot install it when I visit https://demo.photoprism.org. There does not appear to be a service worker available and I think that's why. I'm not sure if this is a feature that hasn't been developed and launched yet or the service worker broke somehow

**Comment by lastzero** [comment]
We don't use service workers at the moment, but there is a `manifest.json` and you can just add it to your home screen (we plan to add a dialog for this later, so that the feature is more visible).

**Comment by lastzero** [comment]
![Uploading Screenshot_20210107_191233.jpg…]()

**Comment by benmccann** [comment]
Ah, okay. I think it's technically not a PWA without a service worker, so I was confused by the use of that term. But I did find the "Add to Home screen" button on mobile and that works for me. Thanks for the clarification!

**Comment by lastzero** [comment]
Not that I am aware of. We can add a worker later, if that helps.

**Comment by lastzero** [comment]
@benmccann Just added a service worker for you :bouquet:

**Comment by benmccann** [comment]
I get the install prompt in Chrome now, which is kind of cool. Screenshot attached

![Screenshot_20210108-104544](https://user-images.githubusercontent.com/322311/104052769-0b875300-519f-11eb-8872-ca298b84d00c.png)

**Comment by lastzero** [comment]
You're welcome!

**Comment by Sherlock1979** [comment]
I don't have the install prompt. Also in the Chrome menu I still see "Add to Home Screen" as opposed to "Install App".

I have forced stopped Chrome, emptied cache and reloaded the Photoprism page but did not help.

PhotoPrism® 210111-cc05c430-Linux-x86_64
Android on Samsung A51.

**Comment by lastzero** [comment]
Are you loading photoprism via https using a valid certificate?

**Comment by Sherlock1979** [comment]
Yes. Nginx reverse proxy using  let's encrypt certificate. (Talking to Photoprism through http.)

**Comment by lastzero** [comment]
@Sherlock1979 Still having issues? Maybe clearing the cache helps:

*On Desktop, the manifest is not automatically updated, but this is planned for a future update.*

https://web.dev/customize-install/#criteria

**Comment by benplumley** [comment]
I came across this issue while trying to figure out why I couldn't install the PWA. For me, putting the server behind Caddy and setting up HTTPS was enough to make the "Install" button to appear in Firefox and Chrome's menus.

The information at https://docs.photoprism.app/user-guide/pwa#__tabbed_1_2 is slightly wrong - it says "Click Add to Home screen" to install the PWA. This actually just adds a bookmark, which opens in a full browser when tapped. When the criteria for PWA are met, there will be a separate button called "Install" underneath "Add to Home screen" which installs the PWA.

**Comment by lastzero** [comment]
Yes, HTTPS is usually required unless you run it on the same host (if you install the PWA on a desktop computer). On Android, the results of "Install" and "Add to homescreen" are essentially the same.... I can't see any difference in functionality, only the icons are different and I think with "install" you can't edit the app name.

**Comment by lastzero** [comment]
@graciousgrey will update our docs accordingly 👍

**Comment by benplumley** [comment]
> On Android, the results of "Install" and "Add to homescreen" are essentially the same

In Firefox for Android, the results are quite different - Install installs a thing that has no browser chrome, whereas "Add to homescreen" adds a bookmark that opens in a regular browser tab with full browser chrome. I'd describe the former as a PWA and the latter as a bookmark. Chrome ~~might~~ does work differently though, they seem to be equivalent there.

**Comment by lastzero** [comment]
I would assume that PWAs are always opened with Chrome while bookmarks will be opened in the browser that added them. Thus the difference if you don't use Chrome as your main browser. On iOS, even Chrome uses the Safari engine. Apple enforces this, they have no choice.

**Comment by benplumley** [comment]
> I would assume that PWAs are always opened with Chrome

This doesn't seem to be the case on my phone (Mi 9T with MIUI12) - opening the PWA originally installed from Firefox, then opening its "App info" page from the switcher, opens the app info page for Firefox.

**Comment by graciousgrey** [comment]
Docs have been updated: https://docs.photoprism.app/user-guide/pwa/#__tabbed_1_2 :)

**Comment by tmzhuang** [comment]
FYI I was prompted to install the PWA when I visited my photoprism instance using Chrome but not for Firefox.

**Comment by lastzero** [comment]
@tmzhuang That's quite possible, because Firefox is neither the native browser on Android (developed by Google) nor on iOS (developed by Apple). If you know of anything WE can do to change that, please let us know! 👍

**Comment by tmzhuang** [comment]
> @tmzhuang That's quite possible, because Firefox is neither the native browser on Android (developed by Google) nor on iOS (developed by Apple). If you know of anything WE can do to change that, please let us know! 👍

At the risk of sounding entitled, would appreciate a mention of it in the docs. I didn't know about the PWA until I accessed the site using my wife's phone since she uses Chrome and I use Firefox. The PWA feels better to use than in the browser on Android. I found this issue by Googling "prompted to install photoprism app on chrome but not Firefox".

Not a huge deal though and I really appreciate all your hardwork on this piece of great software :)
